File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] arr, int k) {
        int n=arr.length;
        k = k%n;
        doReverse(0,n-1,arr);
        doReverse(0,k-1,arr);
        doReverse(k,n-1,arr);
        
    }

    public void doReverse(int s, int e, int[] arr) {
        while (s < e) {
            int temp = arr[s];
            arr[s] = arr[e];
            arr[e] = temp;
            s++;
            e--;
        }
    }
}

File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        
        arr = []
        n = len(nums)
        k %= n
        for i in range(0, k, 1) :
            arr.append(nums[i])
        
        for i in range(n) :
            arr.append(nums[(i+k)%n])
            nums[(i+k)%n] = arr.pop(0)
    
        
        